invasion/enemy.py

```diff
@@ -182,60 +169,70 @@ def hurt(self, player: Member, dmg_type: str, damage: int):
             return
         is_bomb = dmg_type == BOMB_DMG_TYPE
         pid = player.id
-        # dmg as an arg for purchasable bombs
         if (not is_bomb) and dmg_type not in self.hurt_by:
-            # miss penalty
-            self.hurt_mult[pid] = self.hurt_mult.get(pid, 1) * .9
+            self.hurt_mult[pid] = self.hurt_mult.get(pid, 1) * 0.9
             return 0
         hurt_mult = 1 if is_bomb else self.hurt_mult.get(pid, 1)
-        dmg = damage * hurt_mult * max(1 - (self.armor+self.added_armor), 0)
+        dmg = damage * hurt_mult * max(1 - (self.armor + self.added_armor), 0)
         self.health -= dmg
         if self.health <= 0:
             self.health = 0
         self.attacked_by[player.id] = self.attacked_by.get(player.id, 0) + min(self.health, dmg)
         if not is_bomb:
-            # make future hits weaker
-            self.hurt_mult[pid] = self.hurt_mult.get(pid, 1) * .75
+            self.hurt_mult[pid] = self.hurt_mult.get(pid, 1) * 0.75
         else:
             self.bombed_by[player.id] = self.bombed_by.get(player.id, 0) + 1
         return dmg
 
-    def attack(self):
-        self.state = LOSE_STATE
-        return self.penalty_mult
-
     def tick(self):
         if self.health <= 0:
             self.health = 0
             self.state = WIN_STATE
             return
+
+        # Health-based state transitions
+        health_percentage = self.health_percentage
         
+        if health_percentage > 0.75:
```

Comment:
the phases should be based on the configuration done in the `stats.json` file for each monster. please don't hard-code a single monster's configuration into the code